### ISSUE TITLE
False Positives and False Negatives

### ISSUE THREAD

**Issue by rslim97** [angusleigh/leg_tracker#33]
Dear @angusleigh, I'm very thankful for this repo, it has been very useful in most cases developing our following robot.
However there are certain cases when we observed 'ghost detections' trailing on one side of the target when walking near to a wall, this false positive can then cause unwanted id switches. Also we observed a blind spot in the front left of the robot where the lidar is less likely to not detect the pair of legs. Based on these observations my questions are:
1. How to reduce false positive detections?
2. What are the causes for the blind spot?
I hope I can gain some insights from you, once again thanks for your work.

**Comment by angusleigh** [comment]
Hi @rslim97! Are they similar to the false positives you see in the demo video? https://www.youtube.com/watch?v=6k8y72AQG-Y If so, there's not too much that can be done without significant changes. One option might be to map out the area beforehand and use that to filter out false positives appearing in walls, but this would require modifications to the code.

I haven't seen any blind spots before but if you share a screenshot, I could take a look. In general, the algorithm should not perform differently depending on the angle of the person relative to the lidar scanner. So, it might be an issue with the lidar scanner not returning points properly from that angle?

**Comment by rslim97** [comment]
Hi angusleigh, thank you for your feedback, when I set the detection threshold to a smaller value, i.e. 0.1, I do get detections on said blind spots. I think the lidar is returning points properly, however when a person walk towards the robot at a higher speed before sidestepping the number of points landed back may not be sufficient to form clusters to detect the person with high confidence, hence the low confidence clusters. But that's my guess.

Right now, I've realized the need for trade-offs to be made and setting the detection threshold to a lower value is more helpful although it comes with higher false detections. I've tried to overcome this by using sensor fusion to get more robust detections. However, I still get id changes in a crowded setting. It seems there is still more work to be done to achieve reliable, consistent person tracking, especially in complex, dynamic scenes.
